### PR TITLE
Added start and end dates to users

### PR DESCRIPTION
**Pull request by snipe**
This just adds a start date and end date to users and includes the ability to search/sort on those dates via API.

Screen Shot 2022-10-05 at 4 59 24 PM

Screen Shot 2022-10-05 at 5 08 47 PM

Screen Shot 2022-10-05 at 5 09 42 PM

Fixes FD-10911 and SC-9356

### CODE REVIEW

File: app/Http/Transformers/UsersTransformer.php [hunk 1]
```diff
@@ -30,7 +30,7 @@ public function transformUser(User $user)
                 'username' => e($user->username),
                 'remote' => ($user->remote == '1') ? true : false,
                 'locale' => ($user->locale) ? e($user->locale) : null,
-                'employee_num' => e($user->employee_num),
+                'employee_num' => ($user->employee_num) ? e($user->employee_num) : null,
```
Comment (uberbrady) [inline app/Http/Transformers/UsersTransformer.php:33]: This feels a little unrelated but I guess it's fine.

File: app/Http/Transformers/UsersTransformer.php [hunk 2]
```diff
@@ -43,7 +43,7 @@ public function transformUser(User $user)
                 'state' => ($user->state) ? e($user->state) : null,
                 'country' => ($user->country) ? e($user->country) : null,
                 'zip' => ($user->zip) ? e($user->zip) : null,
-                'email' => e($user->email),
+                'email' => ($user->email) ? e($user->email) : null,
```
Comment (uberbrady) [inline app/Http/Transformers/UsersTransformer.php:46]: Also unrelated, but fine.

File: app/Models/User.php [hunk 3]
```diff
@@ -68,6 +70,16 @@ class User extends SnipeModel implements AuthenticatableContract, AuthorizableCo
         'company_id'   => 'integer',
     ];
 
+
+    protected $dates = [
+        'created_at',
```
Comment (uberbrady) [inline app/Models/User.php:75]: This isn't a 'date' it's a timestamp, so I'm not sure that you want this here. Also this is mostly handled by the Timestamps attribute within various models, so I don't know that redeclaring it here is a good idea.

File: resources/views/users/view.blade.php [hunk 4]
```diff
@@ -343,6 +343,30 @@
                         </div>
                       </div>
 
+                   <!-- start date -->
+                   @if ($user->start_date)
+                   <div class="row">
```
Comment (uberbrady) [inline resources/views/users/view.blade.php:348]: Tiny teensy microscopic nit - can we indent the innards of the `@if` directive?

**Review by uberbrady — APPROVED** [review]
This sounds like it could be a useful feature for a lot of people - and I love the idea that, eventually, we might even be able to take actions based on what's here - like if someone is past their `end_date`, and they have things assigned to them - then that might be something we need to handle, somehow. I had a couple of minor stylistic points I raised, but not enough to block the PR. Thanks for making this, and I can't wait to see what our users do with it!

File: app/Http/Transformers/UsersTransformer.php (hunk 1 above)
Comment (snipe) [reply app/Http/Transformers/UsersTransformer.php:33]: It is, but it was wrong before. :(

File: app/Models/User.php (hunk 3 above)
Comment (snipe) [reply app/Models/User.php:75]: Casting things as dates simply makes them grokable via Carbon without us having to instantiate them separately.

File: resources/views/users/view.blade.php (hunk 4 above)
Comment (snipe) [reply resources/views/users/view.blade.php:348]: I already merged that, but I did apply that indent directly to develop

**Pull request merged**